### PR TITLE
New/xline opeartor init

### PR DESCRIPTION
**Pull request by Phoenix500526**
Please briefly answer these questions:
Remove the rust code. Add ci and pre-commit-config.yaml
* what problem are you trying to solve? (or if there's no problem, what's the motivation for this change?)

* what changes does this pull request make?

* are there any non-obvious implications of these changes? (does it break compatibility with previous versions, etc)

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -4,98 +4,49 @@ on:
   schedule:
     - cron: "00 19 * * *" # run ci periodically at 3 am
   pull_request:
-    branches: [ main ]
+    branches: [main]
 
 env:
-  CI_RUST_TOOLCHAIN: 1.70.0
+  GO_VERSION: 1.21
```
[the remark this answers is not in the document]
Comment (Phoenix500526) [reply .github/workflows/ci.yml:10]: Done

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -4,98 +4,49 @@ on:
...
-          profile: default
-          toolchain: ${{ env.CI_RUST_TOOLCHAIN }}
-          override: true
-      - uses: Swatinem/rust-cache@v2
-      - uses: actions-rs/cargo@v1
-        with:
-          command: check
+    - name: Check format
+      run: |
+        output="$(gofmt -l .)"
```
[the remark this answers is not in the document]
Comment (Phoenix500526) [reply .github/workflows/ci.yml:27]: Done

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+# Build
+# the GOARCH has not a default value to allow the binary be built according to the host where the command
+# was called. For example, if we call make docker-build in a local env which has the Apple Silicon M1 SO
+# the docker BUILDPLATFORM arg will be linux/arm64 when for Apple x86 it will be linux/amd64. Therefore,
+# by leaving it empty we can ensure that the container and binary shipped on it will have the same platform.
+RUN CGO_ENABLED=0 GOOS=${TARGETOS:-linux} GOARCH=${TARGETARCH} go build -a -o manager cmd/main.go
+
+# Use distroless as minimal base image to package the manager binary
+# Refer to https://github.com/GoogleContainerTools/distroless for more details
+FROM gcr.io/distroless/static:nonroot
```
[the remark this answers is not in the document]
Comment (Phoenix500526) [reply Dockerfile:28]: Normally, testing xline-opeartor doesn't require a Docker image; We can directly run `make run` for testing. BTW, this Dockerfile is generated by kubebuilder.

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+# and so that source changes don't invalidate our downloaded layer
+RUN go mod download
+
+# Copy the go source
+COPY cmd/main.go cmd/main.go
+COPY api/ api/
+COPY internal/controller/ internal/controller/
+
+# Build
+# the GOARCH has not a default value to allow the binary be built according to the host where the command
```
[the remark this answers is not in the document]
Comment (iGxnon) [reply Dockerfile:20]: It's just a grammar mistake.

[the remark this answers is not in the document]
Comment (Phoenix500526) [reply Dockerfile]: Fixed

**Review by iGxnon — APPROVED** [review]

**Review by LingKa28 — APPROVED** [review]
LGTM

**Review by LingKa28 — APPROVED** [review]

**Pull request merged**